### DISCUSSION TITLE
Input Date/Input date time

### DISCUSSION THREAD

**Discussion by Lu1sma** [Telefonica/mistica-design#1165] [Proposals]
### Goal

Desde el equipo de diseño de O2 hemos detectado la necesidad de realizar la evolución de los componentes Imput Date/Imput date time. Esta evolución se basa en adherir la exclusión de fechas en el calendario.

### Context

Actualmente los componentes Imput Date/Imput date time nos permiten seleccionar todas las fechas disponibles del calendario.

Hemos detectado un caso de uso que es extrapolable a otros proyectos, y por lo tanto creemos que es relevante mencionarlo. Se trata de que un usuario pueda seleccionar una fecha y/o una hora sin que exista total disponibilidad de selección. El hecho es que no siempre todas los horarios y fechas van a estar disponibles debido a un abanico inmenso de motivos (Disponibilidad en fines de semana, indisponibilidad para ofrecer un servicio en una fecha determinada, cierre de tiendas, disponibilidad colapsada en en día determinado...etc.)

Por todo ello, creemos que puede ser enriquecedor realizar una evolución en este componente.

<img width="560" alt="Captura de pantalla 2023-04-03 a las 11 49 02" src="https://user-images.githubusercontent.com/94112294/229474954-83a12cdf-ee6f-4616-82b2-7c383a09720c.png">

### Impact

Se ofrece una solución que infiere directamente en la sistematización del uso de los imputs de selección de fecha.
Actualmente, se están creando soluciones tipo "walk around" debido a que no existe una solución en el DS que nos permita excluir fechas del calendario.

Nuestro caso en O2 es uno de ellos y recientemente se ha dado en un caso similar para un equipo de Movistar.

### Discussion Questions

- ¿Una buena opción de excluir fechas para el "Imput date" sería mostrar la fechas en estado disabled?
- ¿Para el componente imput date time sería conveniente plantear otro tipo de selección de horarios por rango de fechas?

**Comment by HectorSancho** [comment]
Después de consultarlo internamente tanto desde la parte de diseño como la técnica, por ahora no es viable incorporar este componente a Mística.

Es un componente que a los desarrolladores han estimado que les llevaría 1Q (solo desarrollo). Parece que es muy complejo poder bloquear fechas según disponibilidad, festivos, fines de semana y su integración con el backend.

Desde la parte de diseño; definir todas las casuísticas, las variantes, estados y componentes atómicos necesarios para poder montar el componente, también supondría un esfuerzo que no podemos asumir. (para ejemplificarlo dejo las specs del de [Material 3](https://m3.material.io/components/date-pickers/overview)

Creo que tenéis dos opciones:

- Usar el input y data picker nativo de cada plataforma. Es como lo tenemos ahora en Novum para nuestro flujo de cita en tienda, el usuario selecciona una fecha y se valida desde el backend devolviendo las horas disponibles en el caso de que sea elegible o un mensaje al usuario para que seleccione otra fecha, si ese día no hay citas.

- Desarrollar un componente extended desde vuestro propio equipo. Cosa que desaconsejamos porque tampoco os va a rentar por el esfuerzo y recursos necesarios.

La mayoría de productos digitales que no tienen que ver con el sector viajes o de hoteles no suelen invertir recursos en desarrollar y mantener un componente data picker asociado a un input de fecha. Implementan la solución nativa muy similar a la nuestra con sus limitaciones o alguna solución de terceros.

**Reply by HectorSancho** [reply]
Si tenéis alguna duda del flujo actual, o dudas sobre la UX de la primera opción, nos comentáis sin problema y sacamos un rato para verlo.
